request_test.go

```diff
@@ -233,39 +234,94 @@ func TestNewSerializer(t *testing.T) {
 		},
 		{
 			encoding: encoding.JSON,
+			opts:     RequestOptions{Procedure: "procedure"},
+			want:     encoding.JSON,
+		},
+		{
+			encoding: encoding.Raw,
+			opts:     RequestOptions{Procedure: "procedure"},
+			want:     encoding.Raw,
+		},
+		{
+			encoding: encoding.JSON,
 			wantErr:  errMissingProcedure,
 		},
 		{
 			encoding: encoding.Raw,
 			wantErr:  errMissingProcedure,
 		},
 		{
+			desc:     "Thrift auto detect shouldnt fail to call out procedure",
 			encoding: encoding.Thrift,
-			wantErr:  errMissingProcedure,
+			wantErr:  encoding.ErrSpecifyThriftFile,
+		},
+		{
+			desc:     "Thrift encoder with a thrift file and no service",
+			encoding: encoding.Thrift,
+			opts:     RequestOptions{ThriftFile: validThrift},
+			wantErr: encoding.NotFoundError{
+				Msg:       "no Thrift service specified, specify --method Service::Method, available services:",
+				Available: []string{"Simple"},
+			},
 		},
 		{
+			desc:     "JSON encoder with a thrift file and no service",
 			encoding: encoding.JSON,
-			opts:     RequestOptions{Procedure: "procedure"},
-			want:     encoding.JSON,
+			opts:     RequestOptions{ThriftFile: validThrift},
+			wantErr: encoding.NotFoundError{
+				Msg:       "no Thrift service specified, specify --method Service::Method, available services:",
+				Available: []string{"Simple"},
+			},
 		},
 		{
+			desc:     "Raw encoder with a thrift file and no service",
```

Comment:
hmm what does it mean to use raw encoding with a thrift file? thrift files can only be used with a thrift encoding